### PR TITLE
Paper: aPhyloGeo-Covid: A web interface for phylogeographic analysis of SARS-CoV-2 variation using Neo4j and Snakemake

### PR DESCRIPTION
**Pull request by TahiriNadia**
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.

*Editor*: @cbcunc

*Reviewers*: @cmahapatra97, @EngineerKhan, @ferchaure

### CODE REVIEW

**Comment by deniederhut** [comment]
@scoobies mark paper

**Comment by deniederhut** [comment]
@scoobies mark needs review

**Comment by deniederhut** [comment]
@scoobies assign @cbcunc as editor

**Comment by scoobies** [comment]
Assigned! @cbcunc is now the editor

**Comment by deniederhut** [comment]
@scoobies add @cmahapatra97 to reviewers

**Comment by scoobies** [comment]
@cmahapatra97 added to the reviewers list!

**Comment by deniederhut** [comment]
@scoobies code of conduct

**Comment by scoobies** [comment]
Our CoC: https://www.scipy2023.scipy.org/code-of-conduct

**Comment by scoobies** [comment]
Reports of abusive or harassing behavior may be reported to scipy@enthought.com

**Comment by cbcunc** [comment]
Editor says paper is complete and builds. Awaiting review.

**Comment by deniederhut** [comment]
@scoobies add @EngineerKhan to reviewers

**Comment by scoobies** [comment]
@EngineerKhan added to the reviewers list!

**Comment by deniederhut** [comment]
@scoobies add @ferchaure to reviewers

**Comment by scoobies** [comment]
@ferchaure added to the reviewers list!

File: papers/nadia_tahiri/00_bibderwalt.rst [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+   empower phylogeography research. To optimize the utilization of these resources, we have developed an interactive 
+   analysis platform called aPhyloGeo-Covid, leveraging the capabilities of Neo4j, Snakemake, and Python. This platform enables researchers 
+   to explore and visualize a wide range of diverse data sources specifically relevant to  SARS-CoV-2 for phylogeographic analysis. 
+   The integrated Neo4j database acts as a comprehensive repository, consolidating COVID-19 pandemic-related sequences information, 
+   climate data, and demographic data obtained from public databases, facilitating efficient filtering and organization of input data for 
+   phylogeographical studies. Presently, the database encompasses over 113,774 nodes and 194,381 relationships. Once the input dataset is determined. Additionally, aPhyloGeo-Covid provides a scalable and reproducible phylogeographic workflow for investigating the intricate relationship between geographic features and the patterns of variation in diverse SARS-CoV-2 variants. The code repository of platform is publicly accessible on
```
Comment (ferchaure) [inline papers/nadia_tahiri/00_bibderwalt.rst]: The sentence "Once the input dataset is determined." like that is unrelated to the rest of the paragraph

File: papers/nadia_tahiri/00_bibderwalt.rst [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+:cite:`taylor2020intercontinental` :cite:`aziz2022phylogeography`. To bridge this gap, a novel algorithm applying sliding windows to scan the genetic sequence information related to their climatic conditions was developed by our team :cite:`koshkarov2022phylogeography`. This algorithm utilizes sliding windows to scan genetic sequence information in relation to climatic conditions. Multiple sequences are aligned and segmented into numerous alignment windows based on predefined window size and step size. To assess the relationship between variation patterns within species and geographic features, the Robinson and Foulds metric :cite:`robinson1981comparison` was employed to quantify the dissimilarity between the phylogenetic tree of each window and the topological tree of geographic features.
```
Comment (ferchaure) [inline papers/nadia_tahiri/00_bibderwalt.rst:60]: maybe the editors know a better way to cite multiple works to get something like: [15-18] that looks better that the current way

File: papers/nadia_tahiri/00_bibderwalt.rst [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+
+To enable efficient querying, configuration of analysis parameters, and output generation within the database, a driver object was established using the Neo4j Python driver to establish seamless connections with the Neo4j database. For phylogeographic analysis, a streamlined workflow was implemented using the Snakemake workflow management system, ensuring an efficient and structured analysis process. Moreover, the interactive visualization capabilities offered by the Plotly-dash library were leveraged for data exploration, analysis parameter setting, and interactive visualization of results, enhancing the interpretability and user-friendliness of the platform.
```
Comment (ferchaure) [inline papers/nadia_tahiri/00_bibderwalt.rst]: Here you call it "Plotly-dash" but in other places Dash-Plotly, it's a bit confusing.

File: papers/nadia_tahiri/00_bibderwalt.rst [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+
+Lineage nodes establish connections with Nucleotide and Protein nodes, representing the relationships between lineages and their corresponding genetic sequence data. Moreover, Lineage nodes establish relationships with Location nodes, utilizing the most common occurrence rate as a property. This design empowers researchers to determine the most common countries based on lineage names or search for lineages that were predominant in specific countries during specific time periods. This well-structured and interconnected design within the Neo4j database enhances the ability to explore, analyze, and extract meaningful insights from the integrated phylogeographic dataset.
+
+.. figure:: figure1.png
```
Comment (ferchaure) [inline papers/nadia_tahiri/00_bibderwalt.rst:111]: I know that this is a screenshot... but maybe the font could be increased in a easy way?

File: papers/nadia_tahiri/00_bibderwalt.rst [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+
+Various tools and software were utilized to accomplish these analysis tasks, including Biopython :cite:`cock2009biopython`, raxml-ng :cite:`kozlov2019raxml`, fasttree :cite:`price2009fasttree`, and Python libraries such as robinson-foulds, NumPy, and Pandas. A manuscript for aPhyloGeo-pipeline is available on Github Wiki (https://github.com/tahiri-lab/aPhyloGeo-pipeline/wiki).
```
Comment (ferchaure) [inline papers/nadia_tahiri/00_bibderwalt.rst]: Important: add references for Numpy and Pandas

File: papers/nadia_tahiri/00_bibderwalt.rst [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+As shown in Figure :ref:`fig5` the 14 samples exhibited a range of precipitation from 0 mm/day to 8.57 mm/day with an average of 2.13 mm/day. 
+The specific humidity ranged from 2.44 g/kg to 19.33 g/kg, averaging at 9.77 g/kg. 
+The relative humidity ranged from 45.76% to 94.22%, with an average of 73.17%.
+Compared to other parameters, wind speed variability and sky surface shortwave downward irradiance showed relatively small variations across the samples. 
+The sky surface shortwave downward irradiance ranged from 0.67 kW-hr/m2/day to 7.38 kW-hr/m2/day, with an average of 4.25 kW-hr/m2/day. 
+The wind speed at 10 meters ranged from 1.90 m/s to 6.32 m/s, averaging at 3.24 m/s, while the wind speed at 50 meters ranged from 3.22 m/s to 6.40 m/s with an average of 4.39 m/s
+
+
+
+.. figure:: figure5.png
```
Comment (ferchaure) [inline papers/nadia_tahiri/00_bibderwalt.rst:408]: Did you try with bigger dots or bars? it's tricky see the patters or differences just with little dots far way

File: papers/nadia_tahiri/00_bibderwalt.rst [hunk 7]
```diff
@@ -0,0 +1,449 @@
...
+   
+Our phylogeography-based exploration revealed a notable correlation between mutations in the region [792-940] and precipitation. As a reproducible phylogeographic platform, aPhyloGeo-Covid offers the potential to expand the sample size for further investigation and facilitates the comparability of analysis results.
+
+In addition of correlation analysis between correlation diversity of subfregment of gene and climate condition, we also inferred the ORF1a phylogeny and window regions 792-940 of ORF1a using the RAxML-NG method :cite:`kozlov2019raxml`, and then conducted a detailed horizontal gene transfer-recombination analysis (See Figure :ref:`fig7`) using the HGT-Detection program available on the T-Rex web server :cite:`boc2012t`.
```
Comment (ferchaure) [inline papers/nadia_tahiri/00_bibderwalt.rst]: add somewhere: horizontal gene transfer (HGT)

File: papers/nadia_tahiri/00_bibderwalt.rst [hunk 8]
```diff
@@ -0,0 +1,449 @@
...
+In the second iteration, another HGT event was found from subtree QLL26045 to subtrees QPJ77309, QWP89176, and WAV10885, with an RF of 6 and a bipartition dissimilarity of 3.5. After this iteration, the criteria values were RF = 6 and BD = 3.5.
+In the third iteration, an HGT was identified from subtree UJS65740 to subtrees QLL26045, QPJ77309, QPN02498, QWP89176, UJW68561, and WAV10885, with an RF of 4 and a bipartition dissimilarity of 2.5. The criteria values after this iteration remained RF = 4 and BD = 2.5.
```
Comment (ferchaure) [inline papers/nadia_tahiri/00_bibderwalt.rst]: I personally get confuse with this type of paragraph, maybe a table or something with a structure format could be more clear

**Review by ferchaure — CHANGES_REQUESTED** [review]
Hi Nadia, Amazing and really interesting work.

The only required change is adding the reference for Numpy and Pandas.
I added a few additional comments 


Cheers,
Fer

**Comment by EngineerKhan** [comment]
Same thoughts as @ferchaure. I am not an expert on the topic, so my opinion should be taken with a grain of salt. Here are some minor suggested changes, though:

- In the abstract, "a wide range of" is unnecessary.
- "Throughout the COVID-19 pandemic, researchers......" – this is a new section and can be in a new paragraph.
- "Climate data can be obtained from reputable sources such as NASA/POWER DailyGridded weather" – climate data of? Can add more clarity.
- "The challenges associated with data collection, extraction, and integration have hindered the advancement of phylogeographic research within the field. " – I think a reference can be added to this statement.
- "To tackle these challenges, we employed....." – it can also be taken in a new paragraph.
- "To address these challenges, we devised a phylogeographic...." should also be a new paragraph.
- Reference 26 needs to be improved to reflect the better citation of Pandas, like [this](https://conference.scipy.org/proceedings/scipy2010/pdfs/mckinney.pdf).
- "In the subsequent phylogeographical analysis phase, our modularized Snakemake workflow is utilized to
examine how genetic variation patterns within different SARS-CoV-2 variants align with geographic features." – Comment: This line sounds unnecessary for the introduction. Feel free to disagree.
- "from the SARS-CoV-2 Data Hub [7]" – is it the correct reference?
- "population density by country, positivity rates, vaccination rates, diabetes rates, aging data" – sounds like too much info for a general reader.
- " Moreover, the interactive visualization capabilities offered by the Plotly-dash library" – Suggestion: we can add the Plotly-dash library's reference as well.
- The first paragraph of "Data Integration" section contains a lot of attributes and can have a table to complement it.
- "An interactive platform using Dash-Plotly [27] was developped" – please mind the typo.
- "The integration of the Dash platform with the Neo4j graph database allows for seamless retrieval of pertinent data from interconnected nodes based on user-provided keywords related to lineages or locations." – minor: "the" before "seamless retrieval."
- "The aPhyloGeo-Covid offers two distinct approaches for selecting input datasets" – Suggestion: I think these two options would be better off in 2 subheadings rather than 2 numbered sentences.
- "Subsequently, the Neo4j database was queried to identify lineages" – "was" or "is"?
- QUERY: Can we caption the code snippets? @scoobies 
- Suggestion: Fig 2 and 3 can be separated.
- "and the Neighbor-Joining clustering method [31]. " – don't think [31] is the right reference for that method.


Some general observations:
- Majority of references are review/comparative studies.
- At times, it gets verbose.
- Too many figures (numbers) within the text. Some of them can be better represented in tables.
- Is page limit (ref included) 8 or 9? @scoobies

**Comment by cmahapatra97** [comment]
Hi Nadia, 

Very interesting tool development to study SARS-CoV-2 variation. 
I suggest adding some lines on the pros of this model regarding other developed tools based on Neo4J. For example see Laddada W, Zanni-Merk C, Soualmia LF. Analyzing SARS-CoV-2 Sequence Patterns by Semantic Trajectories. Stud Health Technol Inform. 2022 Jun 29;295:197-200. doi: 10.3233/SHTI220696. PMID: 35773842.

Cheers,
Chit

File: papers/nadia_tahiri/00_bibderwalt.rst (hunk 1 above)
Comment (TahiriNadia) [reply papers/nadia_tahiri/00_bibderwalt.rst]: Thank you, @ferchaure , for your comment and suggestions. We agree, and the appropriate modifications have been incorporated by deleting the sentence.

File: papers/nadia_tahiri/00_bibderwalt.rst (hunk 2 above)
Comment (TahiriNadia) [reply papers/nadia_tahiri/00_bibderwalt.rst:60]: Once again, thank you for raising this point, with which we agree. While we may not know how to accomplish it ourselves, we believe that @deniederhut t or @cbcunc might possess the knowledge and expertise to do so.

File: papers/nadia_tahiri/00_bibderwalt.rst (hunk 3 above)
Comment (WanlinLi2021) [reply papers/nadia_tahiri/00_bibderwalt.rst]: Thank you for bringing this issue to my attention. We replaced "Plotly-dash" with "Dash-Plotly" to maintain consistency throughout the article.

File: papers/nadia_tahiri/00_bibderwalt.rst (hunk 4 above)
Comment (WanlinLi2021) [reply papers/nadia_tahiri/00_bibderwalt.rst:111]: Thank you. We acknowledge that utilizing screenshots does not meet the desired image quality standards, lacks reader-friendliness, and fails to highlight key information effectively. Consequently, we have considered your suggestion and have re-created the images using Drawio to enhance the overall quality.

File: papers/nadia_tahiri/00_bibderwalt.rst (hunk 5 above)
Comment (WanlinLi2021) [reply papers/nadia_tahiri/00_bibderwalt.rst]: Thank you for bringing this issue to my attention. I appreciate your feedback. In response to your suggestion, we have now included references for Numpy and Pandas in the article.

File: papers/nadia_tahiri/00_bibderwalt.rst (hunk 7 above)
Comment (WanlinLi2021) [reply papers/nadia_tahiri/00_bibderwalt.rst]: Thank you for your suggestion. Yes, it is essential to clarify abbreviations before using them consistently throughout the text. In response to your feedback, we have added "horizontal gene transfer (HGT)" in the sentence where we initially introduced the concept of horizontal gene transfer.

**Comment by TahiriNadia** [comment]
@scoobies build paper

**Comment by scoobies** [comment]
building paper...

**Comment by scoobies** [comment]
{"data"=>{"info"=>"Build for paper tahiri-lab-2023 scheduled.  Note that builds are only executed if the current build attempt is more than 5 minutes old."}, "status"=>"success"}

File: papers/nadia_tahiri/00_bibderwalt.rst (hunk 8 above)
Comment (WanlinLi2021) [reply papers/nadia_tahiri/00_bibderwalt.rst]: Thank you for the observation. We greatly appreciate it. To ensure conciseness and clarity, we have revised the paragraph concerning transfer-recombination analysis (HGT). Additionally, we have included a table to enhance the clarity of the information presented.

File: papers/nadia_tahiri/00_bibderwalt.rst (hunk 6 above)
Comment (WanlinLi2021) [reply papers/nadia_tahiri/00_bibderwalt.rst:408]: Thank you for your suggestion. We have improved the quality of this figure by increasing the size of the dots.

**Comment by TahiriNadia** [comment]
> Same thoughts as @ferchaure. I am not an expert on the topic, so my opinion should be taken with a grain of salt. Here are some minor suggested changes, though:
> 
> * In the abstract, "a wide range of" is unnecessary.
> * "Throughout the COVID-19 pandemic, researchers......" – this is a new section and can be in a new paragraph.
> * "Climate data can be obtained from reputable sources such as NASA/POWER DailyGridded weather" – climate data of? Can add more clarity.
> * "The challenges associated with data collection, extraction, and integration have hindered the advancement of phylogeographic research within the field. " – I think a reference can be added to this statement.
> * "To tackle these challenges, we employed....." – it can also be taken in a new paragraph.
> * "To address these challenges, we devised a phylogeographic...." should also be a new paragraph.
> * Reference 26 needs to be improved to reflect the better citation of Pandas, like [this](https://conference.scipy.org/proceedings/scipy2010/pdfs/mckinney.pdf).
> * "In the subsequent phylogeographical analysis phase, our modularized Snakemake workflow is utilized to
>   examine how genetic variation patterns within different SARS-CoV-2 variants align with geographic features." – Comment: This line sounds unnecessary for the introduction. Feel free to disagree.
> * "from the SARS-CoV-2 Data Hub [7]" – is it the correct reference?
> * "population density by country, positivity rates, vaccination rates, diabetes rates, aging data" – sounds like too much info for a general reader.
> * " Moreover, the interactive visualization capabilities offered by the Plotly-dash library" – Suggestion: we can add the Plotly-dash library's reference as well.
> * The first paragraph of "Data Integration" section contains a lot of attributes and can have a table to complement it.
> * "An interactive platform using Dash-Plotly [27] was developped" – please mind the typo.
> * "The integration of the Dash platform with the Neo4j graph database allows for seamless retrieval of pertinent data from interconnected nodes based on user-provided keywords related to lineages or locations." – minor: "the" before "seamless retrieval."
> * "The aPhyloGeo-Covid offers two distinct approaches for selecting input datasets" – Suggestion: I think these two options would be better off in 2 subheadings rather than 2 numbered sentences.
> * "Subsequently, the Neo4j database was queried to identify lineages" – "was" or "is"?
> * QUERY: Can we caption the code snippets? @scoobies
> * Suggestion: Fig 2 and 3 can be separated.
> * "and the Neighbor-Joining clustering method [31]. " – don't think [31] is the right reference for that method.
> 
> Some general observations:
> 
> * Majority of references are review/comparative studies.
> * At times, it gets verbose.
> * Too many figures (numbers) within the text. Some of them can be better represented in tables.
> * Is page limit (ref included) 8 or 9? @scoobies


Thank you, @EngineerKhan, for your comment and suggestions. Below are the answers to your questions.


✅ In the abstract, "a wide range of" is unnecessary. 

Thank you for your suggestion. We agree that the phrase "a wide range of" is redundant. Therefore, we have removed it from the text. 

✅ "Throughout the COVID-19 pandemic, researchers......" – this is a new section and can be in a new paragraph. 

We appreciate the suggestion. We also believe that separating it into a new section will enhance clarity. We have taken your advice and incorporated the content into a new section. 

✅ "Climate data can be obtained from reputable sources such as NASA/POWER DailyGridded weather" – climate data of? Can add more clarity. 

Thank you for the suggestion. In order to enhance clarity when referring to the climate data, we have optimized the sentence as follows: "Daily climate data (e.g., humidity, wind speed, and temperature) for each location involved during the pandemic can be obtained from reputable sources such as NASA/POWER DailyGridded Weather." This revision aims to provide more clarity and guidance to readers. 

✅ "The challenges associated with data collection, extraction, and integration have hindered the advancement of phylogeographic research within the field. " – I think a reference can be added to this statement. 

Thank you for your suggestion. We greatly appreciate it. We have taken action by including two additional references to substantiate the information presented. 

✅ "To tackle these challenges, we employed....." – it can also be taken in a new paragraph. 

We appreciate the suggestion. We also believe that separating it into a new section will enhance clarity. We have taken your advice and incorporated the content into a new section, making it easier for readers to locate and understand. 

✅ "To address these challenges, we devised a phylogeographic...." should also be a new paragraph. 

We appreciate the suggestion. We have taken your advice and incorporated the content into a new section. 

✅ Reference 26 needs to be improved to reflect the better citation of Pandas, like [this](https://conference.scipy.org/proceedings/scipy2010/pdfs/mckinney.pdf). 

Thank you for bringing this issue to our attention. We appreciate your feedback. In response to your suggestion, we have now included this reference for Pandas in the article. 

✅ "In the subsequent phylogeographical analysis phase, our modularized Snakemake workflow is utilized to examine how genetic variation patterns within different SARS-CoV-2 variants align with geographic features." – Comment: This line sounds unnecessary for the introduction. Feel free to disagree. 

Thank you for your suggestion. Upon careful consideration, we have decided to retain this line, as it offers a comprehensive overview of the Snakemake section. We believe that including this line will assist readers in understanding the Snakemake portion before delving into the Methodology section. 

✅ "from the SARS-CoV-2 Data Hub [7]" – is it the correct reference? 

Thanks for the reminder. We agree and have added the reference given directly to the SARS-CoV-2 Data Hub website (https://www.ncbi.nlm.nih.gov/labs/virus/vssi/docs/participate/#how-to-cite-us). 

✅ "population density by country, positivity rates, vaccination rates, diabetes rates, aging data" – sounds like too much info for a general reader. 

Thank you for bringing this issue to our attention. In order to facilitate comprehension for general readers, we have divided the long sentences into concise lists. Additionally, we have included Table 1 to provide a clear and accessible representation of the information for easy understanding. 

✅ " Moreover, the interactive visualization capabilities offered by the Plotly-dash library" – Suggestion: we can add the Plotly-dash library's reference as well. 

I sincerely appreciate your suggestion. In response, we have added two references to support further the information provided. One of the references aligns with the suggestion provided by the Plotly Community (source: https://community.plotly.com/t/citing-plotly-and-dash-in-academic-papers/33045). Additionally, we have included a review of Dynamic Dashboards, which include the Dash-Plotly component. 

✅ The first paragraph of "Data Integration" section contains a lot of attributes and can have a table to complement it. 

Thank you for your suggestion. In order to enhance clarity and facilitate understanding, we have incorporated a table into this section. 

✅ "An interactive platform using Dash-Plotly [27] was developped" – please mind the typo. 

Thank you very much for the reminder. We apologize for this mistake. We should have used "developed". We have changed it accordingly. Thanks again. 

✅ "The integration of the Dash platform with the Neo4j graph database allows for seamless retrieval of pertinent data from interconnected nodes based on user-provided keywords related to lineages or locations." – minor: "the" before "seamless retrieval." 

Thank you for this note. We added "the" before "seamless retrieval." 

✅ "The aPhyloGeo-Covid offers two distinct approaches for selecting input datasets" – Suggestion: I think these two options would be better off in 2 subheadings rather than 2 numbered sentences. 

Thank you for this observation. We updated the numbered sentences to subheadings. 

✅ "Subsequently, the Neo4j database was queried to identify lineages" – "was" or "is"? 

Thank you very much for the correction. In the corresponding context in the text, we should have used "is". We have changed it accordingly. Thanks again. 

✅ QUERY: Can we caption the code snippets? [@scoobies](https://github.com/scoobies) 

Thanks for this note and observation. We use exactly the same template given by the SciPy conference (source: https://github.com/scipy-conference/scipy_proceedings/blob/2023/papers/00_bibderwalt/00_bibderwalt.rst) and the template has no caption for code snippets. Thanks. 

 

✅ Suggestion: Fig 2 and 3 can be separated. 

Thank you for your feedback. During the PDF generation process, the rst compiler can automatically reposition figures and tables to optimize spacing. Originally, figure 2 and figure 3 are separated in our Github text. 

✅ "and the Neighbor-Joining clustering method [31]. " – don't think [31] is the right reference for that method. 

Thank you for the note. We understand and agree that the original authors of Neighbor-Joining are Saitou and Nei in 1987(Saitou, Naruya, and Masatoshi Nei, "The neighbor-joining method: a new method for reconstructing phylogenetic trees," Molecular Biology and Evolution,1987). This revised reference will provide a more authoritative and reliable source of information for the topic at hand. 

 

Some general observations: 

✅ Majority of references are review/comparative studies. 

We have added new original references to counter the review/comparative studies. Thanks. 

✅ At times, it gets verbose.  

Thank you for your feedback. We have taken action by rewording certain sentences, splitting others in half, and incorporating additional details and tables. 

✅ Too many figures (numbers) within the text. Some of them can be better represented in tables. 

We have added new tables, and the figures in the text are now easier to understand. Thank you for your attention. 

✅ Is page limit (ref included) 8 or 9? [@scoobies](https://github.com/scoobies) 

Thank you for the reminder. We confirm that the page limit is 8, excluding references. The SciPy conference rule ‘The compiled version of the paper (PDF) should be at most 8 pages, including figures but not including references.’ (Source: https://github.com/scipy-conference/scipy_proceedings/tree/2023).

**Comment by TahiriNadia** [comment]
> Hi Nadia,
> 
> Very interesting tool development to study SARS-CoV-2 variation. I suggest adding some lines on the pros of this model regarding other developed tools based on Neo4J. For example see Laddada W, Zanni-Merk C, Soualmia LF. Analyzing SARS-CoV-2 Sequence Patterns by Semantic Trajectories. Stud Health Technol Inform. 2022 Jun 29;295:197-200. doi: 10.3233/SHTI220696. PMID: 35773842.
> 
> Cheers, Chit

Thank you, @cmahapatra97, for recommending this literature. The article is impressive with the application of Graph theory to mutation tracking. It highlights the advantages of Neo4j in relation analysis. Therefore, we have added a related discussion in the Introduction section.

**Review by ferchaure — APPROVED** [review]

**Comment by cbcunc** [comment]
@scoobies mark pending comment

**Comment by EngineerKhan** [comment]
Thanks a lot, @TahiriNadia for taking my suggestions positively. I was away on vacation, so apologies for the late reply/verification.

**Comment by cbcunc** [comment]
@EngineerKhan does this paper meet your final review approval?

**Comment by cbcunc** [comment]
@scoobies check references

**Comment by scoobies** [comment]
Checking the BibTeX entries failed with the following error: 
```
Lexical or syntactical errors: 

@article{khan2021structurestitle={Structures of SARS-CoV-2 RNA-binding proteins and therapeutic targets},author={Khan, Muhammad Tahir and Irfan, Muhammad and Ahsan, Hina and Ahmed, Abrar and Kaushik, Aman Chandra and Khan, Anwar Sheed and Chinnasamy, Sathishkumar and Ali, Arif and Wei, Dong-Qing},journal={Intervirology},volume={64},number={2},pages={55--68},year={2021},publisher={Karger Publishers},doi=“
@article{ni2023structuraltitle={Structural insights into ribonucleoprotein dissociation by nucleocapsid protein interacting with non-structural protein 3 in SARS-CoV-2},author={Ni, Xincheng and Han, Yinze and Zhou, Renjie and Zhou, Yanmei and Lei, Jian},journal={Communications Biology},volume={6},number={1},pages={193},year={2023},publisher={Nature Publishing Group UK London},doi=“
@article{waterhouse2018swisstitle={SWISS-MODEL: homology modelling of protein structures and complexes},author={Waterhouse, Andrew and Bertoni, Martino and Bienert, Stefan and Studer, Gabriel and Tauriello, Gerardo and Gumienny, Rafal and Heer, Florian T and de Beer, Tjaart A P and Rempfer, Christine and Bordoli, Lorenza and others},journal={Nucleic acids research},volume={46},number={W1},pages={W296--W303},year={2018},publisher={Oxford University Press},doi=“
```

**Comment by cbcunc** [comment]
@TahiriNadia Would you please fix the above BibTex DOI errors? I think they need to go inside curly braces rather than quotes?

**Comment by cbcunc** [comment]
@scoobies check references

**Comment by scoobies** [comment]
Checking the BibTeX entries failed with the following error: 
```
Lexical or syntactical errors: 

@article{khan2021structurestitle={Structures of SARS-CoV-2 RNA-binding proteins and therapeutic targets},author={Khan, Muhammad Tahir and Irfan, Muhammad and Ahsan, Hina and Ahmed, Abrar and Kaushik, Aman Chandra and Khan, Anwar Sheed and Chinnasamy, Sathishkumar and Ali, Arif and Wei, Dong-Qing},journal={Intervirology},volume={64},number={2},pages={55--68},year={2021},publisher={Karger Publishers},doi=“
@article{ni2023structuraltitle={Structural insights into ribonucleoprotein dissociation by nucleocapsid protein interacting with non-structural protein 3 in SARS-CoV-2},author={Ni, Xincheng and Han, Yinze and Zhou, Renjie and Zhou, Yanmei and Lei, Jian},journal={Communications Biology},volume={6},number={1},pages={193},year={2023},publisher={Nature Publishing Group UK London},doi=“
@article{waterhouse2018swisstitle={SWISS-MODEL: homology modelling of protein structures and complexes},author={Waterhouse, Andrew and Bertoni, Martino and Bienert, Stefan and Studer, Gabriel and Tauriello, Gerardo and Gumienny, Rafal and Heer, Florian T and de Beer, Tjaart A P and Rempfer, Christine and Bordoli, Lorenza and others},journal={Nucleic acids research},volume={46},number={W1},pages={W296--W303},year={2018},publisher={Oxford University Press},doi=“
```

**Comment by cbcunc** [comment]
@TahiriNadia We figured out what's wrong with the DOIs. They are quoted with smart quotes instead of straight quotes. Would you please this? We are having a final review today and this is one of the few papers which are not ready.

**Comment by WanlinLi2021** [comment]
> @TahiriNadia We figured out what's wrong with the DOIs. They are quoted with smart quotes instead of straight quotes. Would you please this? We are having a final review today and this is one of the few papers which are not ready.

Thank you very much for the reminder. Sorry for not responding in time. I have made corrections to the QUOTES for these references.

**Comment by cbcunc** [comment]
@scoobies check references

**Comment by scoobies** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/mec.15222 is OK
- 10.1073/pnas.2218012120 is OK
- 10.1111/tbed.14556 is OK
- 10.1111/mec.15991 is OK
- 10.1046/j.1365-294X.2003.02051.x is OK
- 10.12688/wellcomeopenres.16661.2 is OK
- 10.1093/nar/gku1207 is OK
- 10.46234/ccdcw2021.255 is OK
- 10.1016/j.heliyon.2021.e06625 is OK
- 10.1038/s41562-021-01122-8 is OK
- 10.1109/ICDEW.2012.31 is OK
- 10.5220/0006356003510356 is OK
- 10.1046/j.0962-1083.2001.01350.x is OK
- 10.1371/journal.pbio.0020442 is OK
- 10.1038/s41598-020-58743-8 is OK
- 10.3354/esr01188 is OK
- 10.25080/majora-212e5952-018 is OK
- 10.1016/0025-5564(81)90043-2 is OK
- 10.1093/bioinformatics/bty350 is OK
- 10.1093/nar/gkaa434 is OK
- 10.1016/j.future.2023.01.009 is OK
- 10.1038/s41592-021-01254-9 is OK
- 10.1093/bioinformatics/btp163 is OK
- 10.3153/AR19009 is OK
- 10.1145/1900008.1900067 is OK
- 10.1093/database/baab026 is OK
- 10.1007/978-1-4842-0022-3_9 is OK
- 10.1093/molbev/mst010 is OK
- 10.1016/j.sbi.2006.04.004 is OK
- 10.1007/s00453-007-9116-4 is OK
- 10.1093/oxfordjournals.molbev.a040454 is OK
- 10.1093/bioinformatics/btz305 is OK
- 10.1093/molbev/msp077 is OK
- 10.1007/978-3-030-78829-2_9 is OK
- 10.1159/000513686 is OK
- 10.1038/s42003-023-04570-2 is OK
- 10.1093/nar/gky427 is OK
- 10.1093/nar/gks485 is OK
- 10.1016/S0092-8674(00)00175-6 is OK
- 10.1186/s12862-020-01732-2 is OK
- 10.1093/nar/gkr735 is OK
- 10.1109/MCSE.2011.37 is OK
- 10.1007/978-1-4842-0241-8_5 is OK
- 10.1111/jbi.14481 is OK
- 10.1016/j.ympev.2011.12.007 is OK
- 10.1093/nar/gkw1065 is OK
- 10.25080/Majora-7ddc1dd1-012 is OK
- 10.25080/Majora-92bf1922-00a is OK
- 10.3233/SHTI220696 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by cbcunc** [comment]
@scoobies mark ready

**Comment by TahiriNadia** [comment]
> @TahiriNadia Would you please fix the above BibTex DOI errors? I think they need to go inside curly braces rather than quotes?

Thank you, @cbcunc.

**Review by deniederhut — APPROVED** [review]
Thanks @cmahapatra97, @EngineerKhan, and @ferchaure for your help getting this paper ready for publication! And thanks to @TahiriNadia for joining us at SciPy this year (and many other years)!!!

giphy-1806883251

**Pull request merged**